test/hotspot/jtreg/vmTestbase/nsk/share/jpda/SocketConnection.java

```diff
@@ -517,7 +517,9 @@ public Object readObject() {
         try {
             return doReadObject();
         } catch (EOFException e) {
-            return null;
+            e.printStackTrace(logger.getOutStream());
+            throw new Failure("Caught EOFException while reading an object from " + name + " connection."
+                    + " Check if debugee process is exited (crashed, killed) before responded.\n\t");
```

Comment:
Nit: It feels like this line of error message can be improved but I do not have a suggestion at the moment.